wip-0004.md

```diff
@@ -0,0 +1,166 @@
+
+The scope of this document is to propose a solution for an efficient verification of the signatures of the ARS members in a smart contract platform like ethereum. BLS signatures allow us to aggregate signatures provided by the ARS and provide an aggregated signature to a smart contract that can be validated with a single verification. Undoubtedly, such signature and aggregation might have severe consequences in the Witnet chain in terms of efficiency and latency. The following sections propose a solution that minimizes both.
+
+## Proposal 
+
+### Actors in the system
+
+This section describes the different actors that will take part on the protocol:
+
+- **ARS members**: Members who recently participated in the Witnet protocol and that are representative of the current reputation distribution.
+
+- **Disenfranchised nodes**: Nodes who have participated at some point in the Witnet protocol but that are no longer active.
+
+- **Bridge nodes**: Nodes that are in charge of relaying information from Witnet to Ethereum and viceversa. They do not necessarily to actively participate in the Witnet protocol.
+
+### The protocol
+
+This section defines the responsabilities that each of the aforementioned will have in the protocol.
+
+#### Checkpoint based ARS member voting
+
+As already mentioned before, the ARS is a representation of the current reputation distribution in Witnet. Therefore we can expect at least 2/3s of such group to honestly report the current tip of the *Witnet chain*. However, voting for the tip of the chain in every single epoch might incur in non-affordable computational costs. Therefore, we propose to vote at specific chetheckpoints, i.e., every time *epoch mod checkpoint = 0*. Note that the vote should include information about all the block headers that were consolidated in the last *checkpoint* epochs. We will refer to the vote emitted every *checkpoint* as a **superblock**. Currently a block header in Witnet looks like:
```

Comment:
```suggestion
As already mentioned before, the ARS is a representation of the current reputation distribution in Witnet. Therefore we can expect at least 2/3s of such group to honestly report the current tip of the *Witnet chain*. However, voting for the tip of the chain in every single epoch might incur in non-affordable computational costs. Therefore, the proposal is to vote at specific checkpoints, i.e., every time *epoch mod checkpoint = 0*. Note that the vote should include information about all the block headers that were consolidated since last *checkpoint*. We will refer to the vote emitted every *checkpoint* as a **superblock**.
Currently a block header in Witnet looks like:
```